Mage.Sets/src/mage/cards/m/MasterOfCeremonies.java

```diff
@@ -0,0 +1,178 @@
+            List<String> opponentsVotes = vote.getVotes(opponentId);
+            for (String opponentsVote : opponentsVotes) {
+                switch (opponentsVote) {
+                    case "Money": // You and that player each create a Treasure token.
+                        Token treasureOpponent = new TreasureToken();
+                        treasureOpponent.putOntoBattlefield(1, game, source, opponentId);
+
+                        Token treasureController = new TreasureToken();
+                        treasureController.putOntoBattlefield(1, game, source, controller.getId());
+
+                        break;
+                    case "Friends": // You and that player each create a 1/1 green and white Citizen creature token.
+                        Token citizenOpponent = new CitizenGreenWhiteToken();
+                        citizenOpponent.putOntoBattlefield(1, game, source, opponentId);
+
+                        Token citizenOwner = new CitizenGreenWhiteToken();
+                        citizenOwner.putOntoBattlefield(1, game, source, opponentId);
+
+                        break;
+                    case "Secret": // You and that player each draw a card.
+                        Effect drawEffectOpponent = new DrawCardTargetEffect(1);
+                        drawEffectOpponent.setTargetPointer(new FixedTarget(opponentId));
+                        drawEffectOpponent.apply(game, source);
+
+                        Effect drawEffectController = new DrawCardTargetEffect(1);
+                        drawEffectController.setTargetPointer(new FixedTarget(controller.getId()));
+                        drawEffectController.apply(game, source);
+
+                        break;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    private MasterOfCeremoniesChoiceEffect(final MasterOfCeremoniesChoiceEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public MasterOfCeremoniesChoiceEffect copy() {
+        return new MasterOfCeremoniesChoiceEffect(this);
+    }
+}
+
+class MasterOfCeremoniesVote extends VoteHandler<String> {
+
+    @Override
+    protected Set<String> getPossibleVotes(Ability source, Game game) {
+        return new LinkedHashSet<>(Arrays.asList("Money", "Friends", "Secrets"));
+    }
+
+    @Override
+    protected String playerChoose(String voteInfo, Player player, Player decidingPlayer, Ability source, Game game) {
+        // Don't show a choose diagram to the ownder of the ability
+        UUID ownerId = source.getSourceId();
+        if (ownerId.equals(decidingPlayer.getId())) {
+            return "";
+        }
+
+        // Create the chooser
+        ChoiceImpl chooseOutcome = new ChoiceImpl();
+        chooseOutcome.setMessage("Choose money, friends, or secrets");
+        chooseOutcome.setChoices(new LinkedHashSet<>(Arrays.asList("Money", "Friends", "Secrets")));
+
+        decidingPlayer.choose(Outcome.AIDontUseIt, chooseOutcome, game);
```

Comment:
This still needs to be changed. But I'm not sure what to put for it given the AI hint in `vote.doVotes` above